In [1]:
import numpy as np
import pandas as pd 

In [2]:
df = pd.read_csv('sp.csv', sep=';')

In [3]:
pd.set_option('display.max_colwidth', None)

In [4]:
df['ESTADO'] = df.LINK.apply(lambda x: x.split('/')[4])
df['CIDADE'] = df.LINK.apply(lambda x: x.split('/')[5])
df['BAIRRO'] = df.LINK.apply(lambda x: x.split('/')[6])
df['BANCO'] = df.LINK.apply(lambda x: x.split('/')[7])
df['AGENCIA'] = df.LINK.apply(lambda x: x.split('/')[8])

In [5]:
df_sl = df.drop(['ESTADO','LINK', 'NOME', 'BANCO', 'AGENCIA'], axis=1)

In [6]:
gb = df_sl.groupby(['CIDADE']).count()

In [7]:
gb = gb.T

In [8]:
gb.to_csv('gb.csv')

In [9]:
gb

CIDADE  adamantina  adolfo  aguai  aguas-da-prata  aguas-de-lindoia  \
BAIRRO           8       1      6               3                 6   

CIDADE  aguas-de-santa-barbara  aguas-de-sao-pedro  agudos  alfredo-marcondes  \
BAIRRO                       2                   3       6                  1   

CIDADE  altair  ...  valparaiso  vargem-grande-do-sul  vargem-grande-paulista  \
BAIRRO       1  ...           5                     7                       7   

CIDADE  varzea-paulista  vera-cruz  vinhedo  viradouro  vista-alegre-do-alto  \
BAIRRO                7          2       14          4                     1   

CIDADE  votorantim  votuporanga  
BAIRRO          12           13  

[1 rows x 555 columns]

In [10]:
gb['sao-paulo']

BAIRRO    2724
Name: sao-paulo, dtype: int64

In [11]:
df_sp = df[df['CIDADE'] == 'sao-paulo']

In [12]:
df_sp = df_sp.drop(['LINK', 'NOME', 'ESTADO', 'CIDADE', 'AGENCIA'], axis=1)

In [13]:
df_sp.groupby(['BAIRRO']).count()

BANCO
BAIRRO                   
05110-000               1
1310923                 1
aclimacao               9
agua-branca            19
agua-fria               4
...                   ...
vila-universitaria      2
vila-vera               1
vila-vermelha           1
vila-zelina             5
zona-leste              1

[443 rows x 1 columns]

In [14]:
#pd.set_option('display.max_rows', None)
#pd.set_option('display.max_columns', None)

In [71]:
df_sp.shape[0]

2724

In [69]:
df_sp.groupby(['BAIRRO']).count().sort_values('BANCO', ascending=False)

BANCO
BAIRRO                               
centro                            154
bela-vista                        109
itaim-bibi                         97
pinheiros                          69
santo-amaro                        65
cerqueira-cesar                    64
vila-mariana                       51
indianopolis                       47
perdizes                           43
jardim-paulistano                  42
vila-olimpia                       40
ipiranga                           39
vila-nova-conceicao                37
santana                            37
tatuape                            35
jardim-paulista                    35
consolacao                         32
mooca                              32
bras                               31
tucuruvi                           30
lapa                               29
butanta                            26
vila-leopoldina                    25
jardim-america                     24
liberdade                          23
vila-prudente                      22
mirandopolis                       22
penha-de-franca                    22
pacaembu                           21
cidade-moncoes                     20
paraiso                            20
jabaquara                          19
agua-branca                        19
vila-maria                         18
republica                          18
cambuci                            17
casa-verde                         17
socorro                            16
jardim-europa                      16
sao-miguel-paulista                16
saude                              15
morumbi                            15
higienopolis                       15
brooklin-paulista                  15
vila-clementino                    15
vila-andrade                       15
vila-guilherme                     14
alto-de-pinheiros                  14
vila-regente-feijo                 14
jd-paulistano                      14
belenzinho                         13
vila-buarque                       13
santa-cecilia                      13
vila-carrao                        13
vila-formosa                       12
moema                              12
imirim                             12
barra-funda                        12
sumarezinho                        12
itaim                              12
campo-belo                         12
se                                 12
sto-amaro                          11
cidade-mae-do-ceu                  11
vila-sonia                         11
cidade-jardim                      10
jd-paulista                        10
campos-eliseos                     10
limao                              10
vila-albertina                     10
pari                               10
sao-mateus                         10
chacara-santo-antonio               9
itaim-paulista                      9
itaquera                            9
parque-jabaquara                    9
vila-gomes-cardim                   9
vila-pompeia                        9
vila-romana                         9
aclimacao                           9
sapopemba                           9
pirituba                            8
freguesia-do-o                      8
jardim-aricanduva                   8
vila-gertrudes                      8
jd-america                          8
brooklin                            8
jardim-prudencia                    8
vila-madalena                       8
cidade-dutra                        7
brooklin-novo                       7
vila-nova-cachoeirinha              7
jaguare                             7
vila-matilde                        7
vila-re                             7
luz                                 7
b-vista                             7
cangaiba                            6
penha                               6
capao-redondo                       6
alto-da-mooca                       6
guaianazes                          6
santa-efigenia                      6
mandaqui                            5
vila

In [72]:
df_sp.groupby(['BAIRRO']).count().sort_values('BANCO', ascending=False).T.to_csv('sp_por_bairro.csv')


**São paulo possui (2724) linhas no df mas São Paulo possui (2749) segundo a interface do site base**
- centro tem 154 mas deveria estar com 155 segundo a interface do site base
- ITAIM BIBI está com 97 mas deveria estar com 99 segundo a interface do site base
- 

In [25]:
sorted(df[(df['CIDADE'] == 'sao-paulo')].BAIRRO.unique())

['05110-000',
 '1310923',
 'aclimacao',
 'agua-branca',
 'agua-fria',
 'agua-rasa',
 'alphaville',
 'alphaville-industrial',
 'alto-da-boa-vista',
 'alto-da-lapa',
 'alto-da-mooca',
 'alto-de-pinheiro',
 'alto-de-pinheiros',
 'americanopolis',
 'analia-franco',
 'aricanduva',
 'artur-alvim',
 'b-retiro',
 'b-vista',
 'bairro-do-limao',
 'barra-funda',
 'bela-alianca',
 'bela-vista',
 'belem',
 'belenzinho',
 'bras',
 'brooklin',
 'brooklin-novo',
 'brooklin-paulista',
 'brooklin-pta',
 'brooklini',
 'brooklinpaulista',
 'brooklyn-novo',
 'butanta',
 'c-eliseos',
 'c-m-ceu',
 'c-moncoes',
 'cachoeirinha',
 'cambuci',
 'campo-belo',
 'campo-grande',
 'campo-limpo',
 'campos-eliseos',
 'cangaiba',
 'caninde',
 'capao-redondo',
 'carandiru',
 'casa-grande',
 'casa-verde',
 'casa-verde-alta',
 'cde-moncoes',
 'centro',
 'centro-sao-paulo',
 'cerqueira-cesar',
 'ch-monte-alegre',
 'ch-santo-antonio',
 'chac-itaim',
 'chac-santo-antonio',
 'chac-seis-de-outubro',
 'chacara-california',
 'chac

- **Investigando o erro em centro**

In [27]:
df_centro = df[(df['CIDADE'] == 'sao-paulo') & (df['BAIRRO'] == 'centro')]

In [28]:
df_centro

LINK  \
3934        https://agenciasbanco.com.br/agencias/sp/sao-paulo/centro/spinelli-corretora/0000/   
3935          https://agenciasbanco.com.br/agencias/sp/sao-paulo/centro/vision-corretora/0000/   
3938                 https://agenciasbanco.com.br/agencias/sp/sao-paulo/centro/luizacred/0000/   
3941                     https://agenciasbanco.com.br/agencias/sp/sao-paulo/centro/valor/0000/   
3942           https://agenciasbanco.com.br/agencias/sp/sao-paulo/centro/elite-corretora/0000/   
...                                                                                        ...   
4144        https://agenciasbanco.com.br/agencias/sp/sao-paulo/centro/magliano-corretora/0000/   
4147          https://agenciasbanco.com.br/agencias/sp/sao-paulo/centro/lerosa-corretora/0000/   
4149  https://agenciasbanco.com.br/agencias/sp/sao-paulo/centro/ngo-associados-corretora/0000/   
4152          https://agenciasbanco.com.br/agencias/sp/sao-paulo/centro/guitta-corretora/0000/   
4168              https://agenciasbanco.com.br/agencias/sp/sao-paulo/centro/om-corretora/0000/   

                                  NOME ESTADO     CIDADE  BAIRRO  \
3934  Spinelli Corretora Agência 0000      sp  sao-paulo  centro   
3935    Vision Corretora Agência 0000      sp  sao-paulo  centro   
3938           Luizacred Agência 0000      sp  sao-paulo  centro   
3941               Valor Agência 0000      sp  sao-paulo  centro   
3942     Elite Corretora Agência 0000      sp  sao-paulo  centro   
...                                ...    ...        ...     ...   
4144  Magliano Corretora Agência 0000      sp  sao-paulo  centro   
4147    Lerosa Corretora Agência 0000      sp  sao-paulo  centro   
4149      Ngo Associados Agência 0000      sp  sao-paulo  centro   
4152    Guitta Corretora Agência 0000      sp  sao-paulo  centro   
4168        OM Corretora Agência 0000      sp  sao-paulo  centro   

                         BANCO AGENCIA  
3934        spinelli-corretora    0000  
3935          vision-corretora    0000  
3938                 luizacred    0000  
3941                     valor    0000  
3942           elite-corretora    0000  
...                        ...     ...  
4144        magliano-corretora    0000  
4147          lerosa-corretora    0000  
4149  ngo-associados-corretora    0000  
4152          guitta-corretora    0000  
4168              om-corretora    0000  

[154 rows x 7 columns]

In [31]:
df_centro = df_centro.drop(['LINK', 'NOME', 'ESTADO', 'CIDADE', 'BAIRRO'], axis=1)

In [36]:
centro_groupby = df_centro.groupby(['BANCO']).count()

In [37]:
centro_groupby = centro_groupby.T

BANCO    agk-corretora  aymore  baluarte-corretora  banco-bmfbovespa  \
AGENCIA              1       1                   1                 1   

BANCO    banco-bradesco  banco-cacique  banco-citibank  banco-da-amazonia  \
AGENCIA              25              1               3                  1   

BANCO    banco-do-brasil  banco-hsbc  ...  om-corretora  parmetal  \
AGENCIA               30           6  ...             1         1   

BANCO    shayata-corretora  solidez-corretora  spinelli-corretora  \
AGENCIA                  1                  1                   1   

BANCO    torre-corretora  turiscam-corretora  valor  vision-corretora  \
AGENCIA                1                   1      1                 1   

BANCO    walpires-corretora  
AGENCIA                   1  

[1 rows x 48 columns]

In [44]:
centro_groupby.sort_values('AGENCIA', ascending=False)

AGENCIA
BANCO                            
banco-do-brasil                30
banco-bradesco                 25
itau-unibanco                  18
banco-santander                15
caixa-economica-federal        13
banco-hsbc                      6
banco-safra                     4
banco-citibank                  3
jalves-corretora                1
jbcred                          1
lerosa-corretora                1
levycam-corretora               1
luizacred                       1
magliano-corretora              1
ngo-associados-corretora        1
novinvest-corretora             1
agk-corretora                   1
om-corretora                    1
intercam-corretora              1
shayata-corretora               1
solidez-corretora               1
spinelli-corretora              1
torre-corretora                 1
turiscam-corretora              1
valor                           1
vision-corretora                1
parmetal                        1
fourtrade-corretora             1
hcommcor                        1
guitta-corretora                1
baluarte-corretora              1
banco-bmfbovespa                1
banco-cacique                   1
banco-da-amazonia               1
banco-indusval                  1
banco-maxima                    1
banco-pecunia                   1
banco-provincia                 1
banrisul                        1
concordia-commodities           1
dascam-corretora                1
dibran                          1
direcao                         1
elite-corretora                 1
ffa                             1
aymore                          1
guide-investimentos             1
walpires-corretora              1

In [38]:
centro_groupby.T.to_csv('centro_groupbyT.csv')

**Erro encontrado na linha a baixo os dados estão suplicados no site e no dataframe não**
- **https://agenciasbanco.com.br/agencias/sp/sao-paulo/centro/levycam-corretora/**

In [45]:
df_centro

BANCO AGENCIA
3934        spinelli-corretora    0000
3935          vision-corretora    0000
3938                 luizacred    0000
3941                     valor    0000
3942           elite-corretora    0000
...                        ...     ...
4144        magliano-corretora    0000
4147          lerosa-corretora    0000
4149  ngo-associados-corretora    0000
4152          guitta-corretora    0000
4168              om-corretora    0000

[154 rows x 2 columns]

In [46]:
df_sp

BAIRRO                       BANCO
807   chac-santo-antonio             banco-do-brasil
811   chac-santo-antonio           banco-caterpillar
812     ch-santo-antonio  societe-generale-corretora
813   chacara-california             banco-santander
814   chacara-california              banco-bradesco
...                  ...                         ...
4173          bela-vista              banco-citibank
4174          bela-vista              banco-citibank
4175          bela-vista              banco-citibank
4176          bela-vista              banco-citibank
4177          bela-vista              banco-citibank

[2724 rows x 2 columns]

**investigando o erro em Itaim Bibi**

In [54]:
df_itaim = df[(df['CIDADE'] == 'sao-paulo') & (df['BAIRRO'] == 'itaim-bibi')]
df_itaim = df_itaim.drop(['LINK', 'NOME', 'ESTADO', 'CIDADE', 'BAIRRO'], axis=1)

In [73]:
df_itaim.shape[0]

97

In [60]:
pd.set_option('display.max_rows', None)
pd.set_option('display.max_columns', None)

In [67]:
sorted(df_itaim.BANCO.unique())

['abc',
 'ativa-investimentos',
 'banco-abc',
 'banco-abn-amro',
 'banco-barclays',
 'banco-bmg',
 'banco-bradesco',
 'banco-btg-pactual',
 'banco-caixa-geral',
 'banco-cifra',
 'banco-citibank',
 'banco-credit-suisse',
 'banco-do-brasil',
 'banco-fator',
 'banco-hsbc',
 'banco-intermedium',
 'banco-investimentos-credit-suisse',
 'banco-itau-bba',
 'banco-jp-morgan',
 'banco-kdb',
 'banco-modal',
 'banco-morgan-stanley',
 'banco-safra',
 'banco-santander',
 'banco-ubs',
 'bank-of-americarrill-lynch',
 'bcv-banco',
 'bmg-leasing',
 'bny-mellon',
 'br-partners',
 'br-partners-corretora',
 'brickell',
 'brl-trust',
 'btg-pactual-asset-management',
 'btg-pactual-corretora',
 'caixa-economica-federal',
 'cartos-credito',
 'clear-corretora',
 'credit-suisse',
 'credit-suisse-corretora',
 'credit-suisse-hedging-griffo-corretora',
 'deutsche-bank-banco-alemao',
 'fator-corretora',
 'finacred',
 'foco',
 'fram-capital',
 'gavea',
 'global-exchange-corretora',
 'goldman-sachs',
 'goldman-sachs-c

In [61]:
df_itaim.groupby(['BANCO']).count().sort_values('AGENCIA', ascending=False)

AGENCIA
BANCO                                            
banco-santander                                 7
itau-unibanco                                   5
banco-bradesco                                  5
banco-citibank                                  3
banco-do-brasil                                 3
banco-hsbc                                      2
caixa-economica-federal                         2
itau-corretora                                  1
itau                                            1
intl-fcstone                                    1
intesa-sanpaolo                                 1
industrial-do-brasil                            1
icbc                                            1
haitong-securities-corretora                    1
abc                                             1
haitong-corretora                               1
goldman-sachs-corretora                         1
goldman-sachs                                   1
global-exchange-corretora                       1
gavea                                           1
fram-capital                                    1
foco                                            1
guide-investimentos                             1
jp-morgan                                       1
itauvest                                        1
planner                                         1
ubs-corretora                                   1
terra-investimentos                             1
standard-chartered-bank                         1
slw-corretora                                   1
planner-trustee                                 1
planner-corretora                               1
opportunity                                     1
fator-corretora                                 1
novo-mundo-corretora                            1
morgan-stanley-corretora                        1
merrill-lynch                                   1
kirton-corretora                                1
jpmorgan-chase-bank-national-association        1
jp-morgan-corretora                             1
finacred                                        1
credit-suisse                                   1
deutsche-bank-banco-alemao                      1
banco-credit-suisse                             1
banco-kdb                                       1
banco-jp-morgan                                 1
banco-itau-bba                                  1
banco-investimentos-credit-suisse               1
banco-intermedium                               1
banco-fator                                     1
banco-cifra                                     1
banco-morgan-stanley                            1
banco-caixa-geral                               1
banco-btg-pactual                               1
banco-bmg                                       1
banco-barclays                                  1
banco-abn-amro                                  1
banco-abc                                       1
banco-modal                                     1
banco-safra                                     1
credit-suisse-hedging-griffo-corretora          1
brl-trust                                       1
credit-suisse-corretora                         1
ativa-investimentos                             1
clear-corretora                                 1
cartos-credito                                  1
btg-pactual-corretora                           1
btg-pactual-asset-management                    1
brickell                                        1
banco-ubs                                       1
br-partners-corretora                           1
br-partners                                     1
bny-mellon                                      1
bmg-leasing                                     1
bcv-banco                                       1
bank-of-americarrill-lynch                      1
western-asset-management-company                1

**Erro encontrado na linha a baixo os dados estão duplicados no site e no dataframe não**
- https://agenciasbanco.com.br/agencias/sp/sao-paulo/itaim-bibi/novo-mundo-corretora/
- https://agenciasbanco.com.br/agencias/sp/sao-paulo/itaim-bibi/planner-corretora/

- **Investigando o erro em Cerqueira Cesar**

In [77]:
df_cerqueira_cesar = df[(df['CIDADE'] == 'sao-paulo') & (df['BAIRRO'] == 'cerqueira-cesar')]
df_cerqueira_cesar = df_cerqueira_cesar.drop(['LINK', 'NOME', 'ESTADO', 'CIDADE', 'BAIRRO'], axis=1)
df_cerqueira_cesar.groupby(['BANCO']).count().sort_values('AGENCIA', ascending=False)

AGENCIA
BANCO                                   
banco-santander                       12
banco-bradesco                         8
itau-unibanco                          7
banco-do-brasil                        6
banco-safra                            3
caixa-economica-federal                2
banco-citibank                         2
jsafra-corretora                       1
geracao-futuro-corretora               1
get-money-corretora                    1
advanced-corretora                     1
maxima-corretora                       1
financeira-alfa                        1
pan-mercantil                          1
petra-personal-trader-corretora        1
renascenca                             1
santander-securities-services          1
money-plus                             1
banco-triangulo                        1
fair-corretora                         1
bram-bradesco-asset-management         1
alfa-corretora                         1
banco-societe-generale                 1
banco-jsafra                           1
banco-induscred-investimento           1
banco-financiamento                    1
banco-bonsucesso                       1
banco-bna                              1
banco-alfa-investimento                1
banco-alfa                             1
sol-corretora                          1

In [81]:
sorted(df_cerqueira_cesar.BANCO.unique())

['advanced-corretora',
 'alfa-corretora',
 'banco-alfa',
 'banco-alfa-investimento',
 'banco-bna',
 'banco-bonsucesso',
 'banco-bradesco',
 'banco-citibank',
 'banco-do-brasil',
 'banco-financiamento',
 'banco-induscred-investimento',
 'banco-jsafra',
 'banco-safra',
 'banco-santander',
 'banco-societe-generale',
 'banco-triangulo',
 'bram-bradesco-asset-management',
 'caixa-economica-federal',
 'fair-corretora',
 'financeira-alfa',
 'geracao-futuro-corretora',
 'get-money-corretora',
 'itau-unibanco',
 'jsafra-corretora',
 'maxima-corretora',
 'money-plus',
 'pan-mercantil',
 'petra-personal-trader-corretora',
 'renascenca',
 'santander-securities-services',
 'sol-corretora']